### DISCUSSION TITLE
How to create a kind of ''unlisted'' type of repository?

### DISCUSSION THREAD

**Discussion by noxuz** [community/community#22040] [New to GitHub]
<p>Is there a way to create a public type of repository that is only accesible to the person having the link to it?, but it wont appear on search results at the site, just like in youtube for example for unlisted videos.</p>
<p>Thank you in advance.</p>

**Comment by mpboom** [comment, marked as answer]
<p>No, at the time you can’t create unlisted repositories (altough this probably will be logged as a feature request).</p>
<p>Please note you can, however, make a private repository and then invite people who have read-only or read-write access if you have a paid account.</p>

**Reply by SpicyCatGames** [reply]
You have to invite them while you may not have their email.
You may not want to or be able to wait for a response from them.
What we want is the repo to not be listed on our profiles, yet be accessible to those we want with just a URL.

**Reply by cometlolxd** [reply]
You can create gists.

**Reply by SpicyCatGames** [reply]
> You can create gists.

You can also create tiktok videos. Equally irrelevant.

**Reply by pitcheverlasting** [reply]
How can I turn an existing private repo to a gist?

**Reply by seanabraham** [reply]
@SpicyCatGames it's actually highly relevant since gists can be cloned, pulled, pushed to, etc. like a git repository! So you can get pretty close to the functionality of an unlisted repo w/ an unlisted gist. Nice call @cometlolxd.

**Reply by can-non** [reply]
It’s been over two years, and GitHub still shows no sign of wanting to improve. Here’s my own take on a solution:

Creating a hashed account and a single repository seems to be the only viable workaround within GitHub for now. If you're still concerned, hash the filenames and even obfuscate or minify the content. And if you're still worried after all that, just self-host.

**Reply by AgentEnder** [reply]
> It’s been over two years, and GitHub still shows no sign of wanting to improve. Here’s my own take on a solution:
> 
> Creating a hashed account and a single repository seems to be the only viable workaround within GitHub for now. If you're still concerned, hash the filenames and even obfuscate or minify the content. And if you're still worried after all that, just self-host.

I mean if your creating hashed accounts and stuff this is probably not the right solution anyways? Unlisted repos wouldnt be a secure thing, imho they are more so to be able to handle multi file and history in comparison to lists without taking up a ton of room on your profile.

My take is throw a repo up called "misc" or something and use branches with unrelated history, linking to the specific branch that has XYZ on it

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/noxuz">@noxuz</a>,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list.</p>
<p>Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>
<p>Andrea</p>

**Reply by GregW10** [reply]
That's extremely helpful. Thank you. I know there are others I've talked to who have expressed that this would be good for GitHub to have. Hopefully it becomes a feature in the future!

**Reply by m4heshd** [reply]
It's 2023 and still not a feature. Sad.

**Reply by Arisings** [reply]
plz add this :(

**Reply by kittybtw** [reply]
2024 still not a thing :(

**Reply by mrtechtroid** [reply]
@AndreaGriffiths11 Is there any updates on this? This seems like a very useful feature, where you want to show your soon to be clients/employers what you have been working on, but do not want to make it visible on your public profile, (since not open-source).

**Comment by evgenyneu** [comment]
<p>Unlisted repositories will be very good to have. For example, I want to share a URL to my assignment with my teacher. I don’t want to make my repository public, because other students can just Google it and plagiarize. I don’t want to make my repository private either, because I don’t know exactly who will be marking my assignment, so I don’t know whom to invite. Even if I know the person who was marking my assignment, they might not have a github account. So having an “unlisted” type of repository (similar to unlisted Youtube videos) will be very helpful. It will be nice for those repositories to have a long unguessable URLs, like <a href="https://github.com/USER/73a90acaae2b1ccc0e969709665bc62f" rel="nofollow noopener">https://github.com/USER/73a90acaae2b1ccc0e969709665bc62f</a>.</p>
<p>Thanks. :slight_smile:</p>

**Reply by GregW10** [reply]
Totally agree. Would be great if this were a feature.

**Comment by dutzi** [comment]
<p>+1</p>
<p>Our company has a lot of private repos and some public ones.</p>
<p>We maintain a repo named “userscript” full of useful <a href="https://en.wikipedia.org/wiki/Userscript" rel="nofollow noopener">userscripts</a> our teams can use. </p>
<p>Some of these scripts are useful to people outside of R&amp;D, and having them create a GitHub account to access them is a deal-breaker.</p>
<p>We made this repo public so everyone in the company can use it, but now it appears on our company’s GitHub page.</p>
<p>We put a warning in the readme, but having an option to completly hide it from that list would be very useful.</p>
<p>Cheers,</p>
<ul>
<li>Eldad</li>
</ul>

**Comment by DMCK96** [comment]
<p>Currently going through this exact situation and surprised it’s still not a feature.</p>
<p>I’m having to keep it private till the day of the deadline, at which point I’ll make the repo public, but it’s something I really shouldn’t have to do. </p>

**Comment by airtower-luna** [comment]
<p>For smaller things like school projects using secret gists (see <a href="https://gist.github.com/" rel="nofollow noopener">https://gist.github.com/</a>) may be an option. They are git repositories, and according to the documentation hidden from search engines.</p>

**Comment by aangelopoulos** [comment]
<p>+1, this feature should be added</p>

**Comment by mbwangfpdc** [comment]
<p>+1, this is great for sharing simple things with collaborators without GitHub.</p>

**Comment by virajwadhwa** [comment]
<p>+1, interviews are asking for a direct link to code only, and I cannot share school project code publicly!</p>

**Comment by samleibowitz** [comment]
<p>+1. This seems like the perfect feature for use cases where you want to share code of a “not ready for prime time” nature (job interviews, school projects, etc).</p>

**Comment by scabana** [comment]
<p>+1, was looking for this as well. Would be nice to have this to prevent polluting one’s repos, but link to it from an issue as repro steps in another repo.</p>

**Comment by okletsgo-rivas** [comment]
<p>+1, this would be very useful for interviews / job applications to not leak proprietary source code.</p>

**Comment by realityworks** [comment]
<p>+1 This is very important!</p>

**Comment by globglob3D** [comment]
<p>+1 this is very useful</p>

**Comment by brandonhs** [comment]
<p>I would love this feature</p>

**Comment by ghost** [comment]
<p>I too would also like this feature - would be great for classes where you need to share your work but you don’t necessarily want everyone who clicks on your profile to see your work</p>

**Comment by alexclewontin** [comment]
<p>+1, this would be super useful</p>

**Comment by R-N** [comment]
+1 
Seems like still not a thing in 2022. I'd love to be proven wrong though.

**Comment by Aholicknight** [comment]
Even tho this discussion was created a couple of years ago, people can still benefit from this (myself included). Can't wait for this feature to be added to GitHub!

**Comment by aangelopoulos** [comment]
I agree, I'm eagerly awaiting the moment GitHub releases this option.

**Comment by anoushk1234** [comment]
Been two years, when is this coming out?

**Comment by mpboom** [comment]
Due to email notifications being sent out, this topic seems to be getting active again. Please refrain from posting useless replies like "When is this coming?" or "This would be really appreciated".

If you have something new to share (i.e. a workaround), feel free to post a reply. If GitHub has updates to share about the matter, a reply will show up here also. If you're just in support of the idea, give the original post a thumbs up 😄 

That way, you don't litter the public discussion and people who are subscribed to emails about the topic aren't flooded with irrelevant notifications.

**Reply by R-N** [reply]
Won't that just let the topic die silently like two years ago? It's not like it's an open github issue of a feature request. Instead, it's an **answered** question. I doubt they're watching the thumbs up count... Not that I know where to actually support the idea though (which is why I'm here)

**Reply by mpboom** [reply]
Well the original answer explains the desired feature isn't available (at the moment) and the feature request has been logged. The "supportive" replies I talked about are very useless and only litter peoples' inboxes. They help highlighting the issue just as much as just giving a thumbs up (however much that may be).

You do however make a good point about the fact that the current status of the feature request isn't really clear. I've asked the forum staff if there is a way we can make this a bit more clear (for example, by using a label).

**Reply by R-N** [reply]
@mpboom I totally forgot about this. Only revisited thanks to email notification of a new comment. So, did they answer?

**Comment by mightymatth** [comment]
@mpboom, the system of making the right decisions is clearly broken. GitHub is known to make wrong decisions unless there is a [viral community pressure](https://github.com/community/community/discussions/31644) to change something. 

> They [thread replies] help highlighting the issue just as much as just giving a thumbs up (however much that may be).

Thumbs up don't bring attention to anyone. Thread replies with a need for change do, and everyone knows that. Only if people revolt and start spamming, you will start thinking about it. 

A basic user story about this feature is clear as day. People want to make repos public, accessible with a link, without having them listed on GitHub, making user activity invisible in their followers' feeds, undiscoverable by the search engine, etc. The feature is known and available on many platforms like Evernote, Notion, Youtube, etc. despite the difference in a business case.

I understand the security implications and dozens of questions about this feature, but nobody reaches out to talk about it. The majority of people responsible for this don't use GitHub on a daily basis like most of us, craving for this feature, do. 

This thread is a clear example of how community discussions when nobody listens are useless. The majority of responders agreed with the idea, nobody stated any argument against it. They are just waiting for you to start making some actions, but after seeing this old thread with multiple '+1's comments, they get discouraged to do anything and just go away.

I'm not blaming you personally, but if anyone wanted to clear the feature out, they would shout out for suggestions to get feedback, make a plan, create a feature and close this issue after 4 years of activity. Communication. Communication before all. There is a number of organizations (e.g. Golang community) that have a healthy procedure for solving feature requests.

**Comment by ChrisSimoniMBT** [comment]
This would be really appreciated. When is this coming?

**Comment by SpicyCatGames** [comment]
Having a way to share an "invite link" for private repositories would solve some of the problems that unlisted repository would solve. We do not have that either. If you don't know who you need to see your repo, there is no option but to make it public.

**Reply by R-N** [reply]
AFAIK pending invitations expire within 7 days. If there were an invite link, I think it'd behave the same. If we don't know who'd see the repo, we **likely** won't know when they would see it either. Might exceed the time limit.

**Reply by R-N** [reply]
Also, can a free account restrict the collaborator access to read only?

**Reply by SpicyCatGames** [reply]
@R-N I believe so, there are other limitation. But in regards to your first reply, you're missing the point. Let me give you an example, I work a decent job as a programmer but I'm not done with university. I also have friends whom I respect. I don't want them to see the crap I made for university, or anyone who can just visit my profile, so it will be public with no restriction, other than the fact that it will have no SEO or be visible on my profile.

Check youtube unlisted videos to get the idea.

**Comment by shridhardamle** [comment]
Public Unlisted would be a great feature. I would like to distribute some libraries via package managers ( SPM etc. ) - which need repo to be public. But I have no need to make that repo listed.

**Comment by jzelAdmin2006** [comment]
**Please GitHub: Give this issue more priority**
GitHub is a platform that has revolutionized the way developers and software teams collaborate and share their work with the world. While GitHub provides different options for visibility of a repository such as public, private, and internal, there are still certain scenarios where these options do not suffice. In particular, as we all can see, a lot of users have been requesting a public but unlisted option where they can share their work with specific people without the repository being publicly visible to everyone. This feature has been requested for various reasons, including sharing school projects with teachers without having to make them public to other students, sharing scripts or tools within an organization without it being listed publicly, and more.

Several users have voiced their support for this feature and have shared their use cases, but as of yet, GitHub has not implemented this feature. However, the fact that GitHub has logged this feature request and considered it for implementation is a positive sign. Therefore, it is important for users to keep voicing their opinions and building up pressure for GitHub to implement this feature. As more people support this feature, it is more likely that GitHub will prioritize its implementation.

Additionally, this feature would be extremely useful for organizations and individuals who do not want to make their repositories publicly available but still want to share them with specific people. For example, a company may have internal tools that they want to share with a few clients, but they do not want these tools to be publicly visible on their GitHub page. Similarly, students may want to share their assignments with their teachers without making them public to other students or the internet.

In conclusion, the implementation of the public unlisted visibility feature on GitHub would be a great addition to the platform. It would provide users with more options for visibility and would be particularly useful for individuals and organizations who need to share their work with specific people without making it publicly available to everyone. Therefore, it is crucial for users to continue to support this feature and keep building pressure on GitHub to implement it.

I want to stress that implementing this feature would not only benefit the people of this discussion and me, but also countless other users. I believe that this is not a huge ask, and would not require significant resources or manpower to implement.

Moreover, by implementing this feature, GitHub could demonstrate its commitment to empowering developers and facilitating collaboration. I mean, it's been almost 5 years now, it's basically the least you can do now. This would not only be a great way to improve the user experience, but also to reinforce GitHub's position as a leading platform for software development.

In summary, I strongly encourage the GitHub team to consider adding this feature, as I believe it would greatly enhance the functionality and usability of the platform. I appreciate your time and consideration, and I look forward to seeing the continued evolution of GitHub.

Thank you so much in advance

**Comment by olawalejuwonm** [comment]
Is this available now?

**Reply by SpicyCatGames** [reply]
No

**Comment by KevinNorth** [comment]
I would also appreciate this feature.

**Comment by a-aznar** [comment]
+1

**Comment by vsriram11** [comment]
Any news on this? Really could use unlisted repos as a feature

**Comment by CoffeelessProgrammer** [comment]
+1 for public repos only accessible via url, i.e. unlisted & unsearchable

**Reply by CoffeelessProgrammer** [reply]
Checking in a few months after initial reply and looks like no updates. This feature was requested 5+ years ago, when will it be available for paid plans?

**Comment by sjau** [comment]
+1 here also

**Comment by L0garithmic** [comment]
With the partnership between Github and Cloudflare Pages, I would like my website's code and scripts not publicly available and searchable. Some of the info is private. Please make unlisted repos a thing.

**Comment by Roxcsas** [comment]
+1

**Comment by nydragon** [comment]
+1

**Comment by Hyrumdrums** [comment]
+1

**Comment by akshhack** [comment]
+1

**Comment by lukepighetti** [comment]
would love unlisted repos so I can make code available to my CI/CD system without having to setup keys etc. Obviously not idea but would work well for my use case and make things a lot easier.

**Comment by donjuanpond** [comment]
+1 from me!

**Comment by HacksPloiter** [comment]
+1 from me as well!

**Comment by Jameswlepage** [comment]
+1

**Comment by abhinav-k29** [comment]
Half a decade and they still haven't worked on this feature, what are developers doing?

**Comment by zaunere** [comment]
This is like a time machine.

**Comment by Ceebox** [comment]
Please add this. I am considering switching away from GitHub to self-hosted.

**Comment by AgentEnder** [comment]
It's not as hood as a real solution, but for hosting issue reproduction this works pretty well: https://craigory.dev/blog/2024-09-11/gh-unlisted-repos/

**Comment by ARHAEEM** [comment]
yaaaay 🎉 
I just got confirmation this feature will be released 100% between Q2&Q3 within 2030-2031. Stay tuned guys! 😂 😂 😂

**Comment by hirasso** [comment]
Just use BitBucket for the repos you don't want to have listed. Having them public there is almost as good as an unlisted repo on GitHub ;)

**Comment by hdadhich01** [comment]
gitfront.io works well

**Comment by MoongirlSeline** [comment]
+1 i want unlisted repos

**Comment by jszklarz** [comment]
+1 for unlisted 😢

**Comment by TTy32** [comment]
+1

**Comment by orchidoris** [comment]
I would like to have an unlisted repo to send my portfolio pet-project to potential employers without getting it exposed to web crawlers and used to train AI. An average employer wouldn't go through the process of being added as a read-only contributor to the repo, and that's why the unlisted functionality would be a great solution for such a scenario. GitHub is the best place for it as everyone knows its' interface.

Quick and easy access by a target audience only is a key here. Please, add it! 🤩

**Comment by dkennell** [comment]
+1. A developer's GH account often serves as a resume, where you don't want your most recently worked-on repos to be random cruft that you were playing around with. However, making your repos fully private limits them in other ways besides just visibility in the GUI.

My use case is this: I'd like to clone my random-cruft experimental repos onto various slapdash servers that I may spin down again soon. I don't want to add SSH keys for these servers to my account, since I'm just running quick experiments - I don't want to have to think about security, appropriate permissions, or having to manage those keys in the future. I'd just like my repo to be publicly accessible via git, but also not have to think about my peers and potential employers seeing these quick and sloppy experiments as my most recently worked-on repo.

Unlisted repos give us a tasteful layer of semi-privacy. Like a little gate on a short picket fence around a garden. It's not about hardcore security here, it's about giving us the tools to just make things nice and presentable. Thanks.

Edit: SourceHut's UI is comically rough looking, but they have an Unlisted option. And by default it gives you one URL for read/write and another for read-only. Works fine, seems like a good place to do quick-and-dirty projects you're not wanting to present to others.

**Comment by mockrounds** [comment]


**Reply by SpicyCatGames** [reply]
oh god what kind of ai garbage is this? it made questions out of this discussion and posted the link here.

**Reply by realityworks** [reply]
> oh god what kind of ai garbage is this? it made questions out of this discussion and posted the link here.

Yes, I've reported it as a spam bot. If you haven't yet, do report it.

**Reply by mockrounds** [reply]
Sorry about the confusion, I have removed the comment as its unrelated.

**Comment by sjau** [comment]
Almost 7 years later.....

**Comment by tonykipkemboi** [comment]
okay, common someone, anyone at GitHub reading this. 
at least tell us the reasons why this might not be added? 
am sure am not the only one resorting to Google Colab for some of this functionality (Python only of course).

**Reply by hirasso** [reply]
They won't. Not enough business value/AI in this feature 😅

**Reply by realityworks** [reply]
> They won't. Not enough business value/AI in this feature 😅

They are firing everyone because they aren't making the money back from AI investment and need to look good on paper... We call it the tech ceiling.

**Comment by projektorius96** [comment]
I'll probably repeat myself again, but unlisted repo would really endow a freelance developer to share the code via links on platforms such as **Buy Me a Coffee**. The feature would be game-changing, and may attract more people going forward with this platform, however I see GitHub core team turned a blind eye on this for unknown reason, sadly... 🥹

**Reply by hlevring** [reply]
Totally agree

**Comment by 0x03B4** [comment]
Well, it has been 8 years since the feature request was "shared with the appropriate teams for consideration.", it seems like they have no intentions of implementing it. :)

**Comment by AgentEnder** [comment]
I know i already posted this... but it really does get you like 98% of the way there.

> It's not as good as a real solution, but for hosting issue reproduction this works pretty well: https://craigory.dev/blog/2024-09-11/gh-unlisted-repos/

**Comment by revoconner** [comment]
I made a github app https://www.github-unlisted.com/ for this exact purpose. Almost a decade old request and no reply from github other than the "feature request". I'm dropping this here in case it's useful. It's free, I don't make any money out of it. 

Would definitely prefer a native feature, though.
